soc/nordic/nrf54h/soc.c

```diff
@@ -29,6 +30,8 @@ LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
 #define HSFLL_NODE DT_NODELABEL(cpurad_hsfll)
 #endif
 
+sys_snode_t soc_node;
```

Comment:
A comment explaining that this is not static because it is also accessed from `power.c` would not hurt.

Reply:
This node really should not be shared in the first place... Instead, add an power_init() function to power.c which requests the power domain, moving the node to power.c and making it static, and call `power_init()` from soc.c

Reply:
or, just move `power_domain_init()` to power.c?

Reply:
This is a bit complicated, because `power.c` is only included in the build for `CONFIG_PM OR CONFIG_POWEROFF`.

Reply:
How about `#ifdef`ed `power_init()` call?